workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Readers/ReaderBase.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using Improbable.Worker;
+using Unity.Entities;
+using UnityEngine;
+using Entity = Unity.Entities.Entity;
+
+namespace Improbable.Gdk.Core
+{
+    public abstract class ReaderBase<TSpatialComponentData, TComponentUpdate>
+        : IReader<TSpatialComponentData>,
+            IReaderInternal
+        where TSpatialComponentData : ISpatialComponentData
+        where TComponentUpdate : ISpatialComponentUpdate<TSpatialComponentData>
+    {
+        protected readonly Entity Entity;
+        protected readonly EntityManager EntityManager;
+
+        protected ReaderBase(Entity entity, EntityManager entityManager)
+        {
+            Entity = entity;
+            EntityManager = entityManager;
+        }
+
+        public Authority Authority
+        {
+            get
+            {
+                if (EntityManager.HasComponent<Authoritative<TSpatialComponentData>>(Entity))
+                {
+                    return Authority.Authoritative;
+                }
+
+                if (EntityManager.HasComponent<AuthorityLossImminent<TSpatialComponentData>>(Entity))
```

Comment:
If you are `AuthorityLossImminent` you will have both the `Authoritative` and `AuthorityLossImminent` tags - as you are strictly still Authoritative - this allows users to not have to duplicate injections in ECS systems.

Should reorder these ifs to reflect that